plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureBlobStore.java

```diff
@@ -79,50 +81,52 @@ public BlobContainer blobContainer(BlobPath path) {
     }
 
     @Override
-    public void delete(BlobPath path) {
-        String keyPath = path.buildAsString();
+    public void delete(BlobPath path) throws IOException {
+        final String keyPath = path.buildAsString();
         try {
-            this.client.deleteFiles(this.clientName, this.locMode, container, keyPath);
+            service.deleteFiles(clientName, container, keyPath);
         } catch (URISyntaxException | StorageException e) {
-            logger.warn("can not remove [{}] in container {{}}: {}", keyPath, container, e.getMessage());
+            logger.warn("cannot access [{}] in container {{}}: {}", keyPath, container, e.getMessage());
+            throw new IOException(e);
         }
     }
 
     @Override
     public void close() {
     }
 
-    public boolean doesContainerExist()
-    {
-        return this.client.doesContainerExist(this.clientName, this.locMode, container);
+    public boolean doesContainerExist() {
+        logger.trace("containerExists({})", container);
+        try {
+            return service.doesContainerExist(clientName, container);
+        } catch (URISyntaxException | StorageException e) {
```

Comment:
I have adjusted it to also throw exception.

I guess the original reasoning was that boolean `exists` should swallow the exception and return false, making code simpler and presumably because if indeed there is a connection problem the exception will be rethrown later, when trying the create/access associated operation. Yet, this is not consistent across the plugin code, and your suggestion is indeed in tone with the surrounding.